canisters/src/vts/src/lib.rs

```diff
@@ -208,6 +211,63 @@ struct AgreementConditions {
     gas_price: String,
 }
 
+#[ic_cdk::update(guard = is_user)]
```

Comment:
> User is not needed this function to execute. We need to execute it every time after data accumulating. If today is a first day of month let's execute it.

so I can delete this line: `#[ic_cdk::update(guard = is_user)]` ?

Reply:
> But do not create two invoices for the same telemetry.

do you mean something like that in create_invoice method?
`if existing_invoice -> return Error::AlreadyExists?`

Reply:
We need to check that for the previous month we already have created invoice and not create new one. So to avoid paying twice.

Yes, you can delete `#[ic_cdk::update(guard = is_user)]` and use this function from telemetry accumulation functions.